mesos-rxjava-core/src/main/java/org/apache/mesos/rx/java/MesosSchedulerClient.java

```diff
@@ -114,4 +138,75 @@ public Subscription sink(@NotNull final Observable<SinkOperation<Send>> spout) {
             .observeOn(Rx.compute())
             .subscribe(subscriber);
     }
+
+    @VisibleForTesting
+    static final class UserAgent {
+        private static final Joiner JOINER = Joiner.on(" ").skipNulls();
+        @NotNull
+        private final List<UserAgentEntry> entries;
+
+        @SafeVarargs
+        public UserAgent(@NotNull final Function<Class<? extends UserAgent>, UserAgentEntry>... entries) {
```

Comment:
Is there a good reason why the construction of the `UserAgentEntry` has to wait until it receives the `UserAgent.class` instance in this constructor? Why that particular class? Even if it has to be that class, why not refer to it in the static factory method? For example:

```java
@NotNull
@VisibleForTesting
static UserAgentEntry userAgentEntryForGradleArtifact(@NotNull final String artifactId) {
    final Properties props = loadProperties(UserAgent.class, String.format("/META-INF/%s.properties", artifactId));
    return new UserAgentEntry(props.getProperty("artifactId", artifactId), props.getProperty("Implementation-Version", "unknown-version"));
}
```